coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -319,7 +319,12 @@ public boolean restore_state(InputObjectState os, int ot) {
                     }
 
                     if (status == LRAStatus.Active) {
-                        status = LRAStatus.Cancelling; // transition from Active to Cancelling
+                        if (getSize(pendingList) == 0 && getSize(preparedList) == 0 && getSize(heuristicList) == 0) {
+                            updateState(LRAStatus.Cancelled);
+                        }
+                        else {
+                            updateState(LRAStatus.Cancelling);
+                        }
```

Comment:
There is a question if we can move this to new code to use the scheduleCancellation function somehow. As I understand it abortLRA causes a locking issue but scheduleCancellation would hopefully be in a different thread.